python/pip.bzl

```diff
@@ -0,0 +1,54 @@
+# Copyright 2017 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""A rule for importing Python dependencies via requirements.txt."""
+
+def _import_impl(repository_ctx):
+  """Core implementation of pip_import."""
+
+  # Add an empty top-level BUILD file.
+  repository_ctx.file("BUILD", "")
```

Comment:
I'm not sure I understand your concern.  This file is being created within the context of the repository, not in the user's main build tree.  I only create this because otherwise skylark rules within the repository cannot be loaded.

I added a better comment to clarify.